nifi-nar-bundles/nifi-prometheus-bundle/nifi-prometheus-reporting-task/src/main/java/org/apache/nifi/reporting/prometheus/PrometheusServer.java

```diff
@@ -0,0 +1,165 @@
+
+    class MetricsServlet extends HttpServlet {
+        private CollectorRegistry nifiRegistry, jvmRegistry;
+        private ProcessGroupStatus rootGroupStatus;
+
+        @Override
+        protected void doGet(final HttpServletRequest req, final HttpServletResponse resp) throws ServletException, IOException {
+            logger.info("PrometheusServer Do get called");
+
+            rootGroupStatus = PrometheusServer.this.context.getEventAccess().getControllerStatus();
+            ServletOutputStream response = resp.getOutputStream();
+            OutputStreamWriter osw = new OutputStreamWriter(response);
+            nifiRegistry = PrometheusMetricsUtil.createNifiMetrics(rootGroupStatus, PrometheusServer.this.applicationId);
+            TextFormat.write004(osw, nifiRegistry.metricFamilySamples());
+
+            if (PrometheusServer.this.sendJvmMetrics == true) {
+                jvmRegistry = PrometheusMetricsUtil.createJvmMetrics(VirtualMachineMetrics.getInstance());
+                TextFormat.write004(osw, jvmRegistry.metricFamilySamples());
+            }
+
+            osw.flush();
+            osw.close();
+            response.flush();
+            response.close();
+            resp.setHeader("Content-Type", TextFormat.CONTENT_TYPE_004);
+            resp.setStatus(HttpURLConnection.HTTP_OK);
+            resp.flushBuffer();
+        }
+    }
+
+    public PrometheusServer(InetSocketAddress addr, ComponentLog logger) throws Exception {
+        PrometheusServer.logger = logger;
+        this.server = new Server(addr);
+
+        this.handler = new ServletContextHandler(server, "/metrics");
+        this.handler.addServlet(new ServletHolder(new MetricsServlet()), "/");
+        this.server.start();
+    }
+
+    public PrometheusServer(int addr, SSLContextService sslContextService, ComponentLog logger) throws Exception {
+        PrometheusServer.logger = logger;
+        this.server = new Server();
+        this.handler = new ServletContextHandler(server, "/metrics");
+        this.handler.addServlet(new ServletHolder(new MetricsServlet()), "/");
+
+        SslContextFactory sslFactory = createSslFactory(sslContextService);
+        HttpConfiguration httpsConfiguration = new HttpConfiguration();
+        httpsConfiguration.setSecureScheme("https");
+        httpsConfiguration.setSecurePort(addr);
+        httpsConfiguration.addCustomizer(new SecureRequestCustomizer());
+
+        ServerConnector https = new ServerConnector(server, new SslConnectionFactory(sslFactory, "http/1.1"),
+            new HttpConnectionFactory(httpsConfiguration));
+        https.setPort(addr);
+        this.server.setConnectors(new Connector[]{https});
+        this.server.start();
+    }
+
+    private SslContextFactory createSslFactory(final SSLContextService sslService) {
+        SslContextFactory sslFactory = new SslContextFactory();
+        sslFactory.setNeedClientAuth(true);
+        sslFactory.setWantClientAuth(false);
```

Comment:
`needClientAuth` and `wantClientAuth` cannot be set to hardcoded values. For example, in the case where the user wants the prometheus metrics endpoint to use https in transport, but not verify the client identity, both would be false. 

There is an open JIRA [NIFI-1652](https://issues.apache.org/jira/browse/NIFI-1652) for allowing these values to be set in the SSLContextService. In the meantime, these need to be inferred or user-configurable. 

One option is inferring `needClientAuth` based on if a truststore was configured in the SSLContext as in [ListenHTTP](https://github.com/apache/nifi/blob/0cb15cfb1a5425704a60db2d49415f745df48b1b/nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/ListenHTTP.java#L244).

An alternative option is to let the user specify this as in [HandleHttpRequest](https://github.com/apache/nifi/blob/0cb15cfb1a5425704a60db2d49415f745df48b1b/nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/HandleHttpRequest.java#L334).

Reply:
Hi @kevdoran Thanks for reviewing. I'll refer HandleHttpRequest code for clientAuth part and submit the changes asap.